addons/web/static/src/model/relational_model/dynamic_group_list.js

```diff
@@ -252,6 +252,15 @@ export class DynamicGroupList extends DynamicList {
             for (const group of groups) {
                 delete configGroups[group.value];
             }
+            const currentGroupIds = [];
+            for (const group of this.groups){
+                currentGroupIds.push(group.value);
+            }
+            for (const key in configGroups){
+                if (!(key in currentGroupIds)){
+                    delete configGroups[key];
+                }
+            }
```

Comment:
```suggestion
            const currentGroupIds = [];
            for (const group of this.groups) {
                currentGroupIds.push(group.value);
            }
            for (const key in configGroups) {
                if (!(key in currentGroupIds)) {
                    delete configGroups[key];
                }
            }
```
if you don't have a linter set up, I would greatly recommend doing that